### DISCUSSION TITLE
Transform Estimation DisplayableObjects

### DISCUSSION THREAD

**Discussion by stka85** [PlusToolkit/PlusLib#1163] [General]
Hi, I use the Clarius C3 HD3 and the Phantom 3.1. For the configuration file I followed the example from https://github.com/PlusToolkit/PlusLibData/blob/master/ConfigFiles/PlusDeviceSet_fCal_Ultrasonix_C5-2_NDIPolaris_fCal3.xml. Can you please explain how the transformation matrices ModelToObject are determined for the Probe and the Phantom? Thank you!

**Comment by Sunderlandkyl** [comment]
It's going to depend on the coordinate system of your stl model.

For the probe, it is converted from LPS to RAS and translated so that the origin of the transudcer is at [0, 0, 0]. This coordinate system goes from the "Model" to ObjectCoordinateFrame "TransducerOrigin". I'm not sure how this was done, but I would guess that the position of the transducer origin was located manually on the model. Possibly in 3D Slicer.

For the phantom coordinate system, the wires are defined based on the model so it is already in the "Phantom" coordinate system. No transformation needs to be applied to go from "Model" to ObjectCoordinateFrame "Phantom"

**Reply by stka85** [reply]
Thank you this helps a lot!
